### PR TITLE
Extend capabilities to summon instances

### PR DESCRIPTION
**Pull request by joroKr21**
* `summonOnly` - summon exactly one instance or fail
 * `summonNone` - ensure there are no instances available

I'm not sure of the names. If that seems useful, I will add some tests.

### CODE REVIEW

File: modules/deriving/src/main/scala/shapeless3/deriving/kinds.scala [hunk 1]
```diff
@@ -185,12 +219,27 @@ object K1 {
...
 
   transparent inline def summonFirst0[T]: Any =
     K0.summonFirst0[T]
 
+  /**
+   * Summon the only given instance `F[U]` from the tuple `T`.
+   * Remaining elements of `T` are guaranteed to not have an instance of `F`.
+   */
+  inline def summonOnly[F[_[_]], T[_], U[_]]: F[U] =
+    K0.summonOnly0[LiftP[F, T]].asInstanceOf[F[U]]
```
Comment (TimWSpence) [inline modules/deriving/src/main/scala/shapeless3/deriving/kinds.scala:237]: Should these methods be extracted to somewhere other than `K0` as they aren't specific to a given kind?

**Review by TimWSpence — APPROVED** [review]
Nice!

File: modules/deriving/src/main/scala/shapeless3/deriving/kinds.scala (hunk 1 above)
Comment (joroKr21) [reply modules/deriving/src/main/scala/shapeless3/deriving/kinds.scala:237]: Idk, `summonFirst0` is already in `K0` and we can't move it now 🤷 
Do you have any better idea? In general I agree and even `summonFirst0` is present multiple times 😬

Comment (joroKr21) [reply modules/deriving/src/main/scala/shapeless3/deriving/kinds.scala]: @TimWSpence I moved them to an object in the `internals` package 👍 
I think that's much better now. Also deprecated the old helpers.

**Pull request merged**